tests/periph_pm/main.c

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2016-2018 Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Power management peripheral test.
+ *
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "periph/pm.h"
+#ifdef MODULE_PERIPH_RTC
+#include "periph/rtc.h"
+#endif
+#include "pm_layered.h"
+#include "shell.h"
+
+#ifdef MODULE_PM_LAYERED
+static int cmd_block(int argc, char **argv);
+#endif
+static int cmd_lowest(int argc, char **argv);
+static int cmd_off(int argc, char **argv);
+static int cmd_reboot(int argc, char **argv);
+#ifdef MODULE_PM_LAYERED
+static int cmd_set(int argc, char **argv);
+static int cmd_unblock(int argc, char **argv);
+#ifdef MODULE_PERIPH_RTC
+static int cmd_unblock_rtc(int argc, char **argv);
+#endif
+#endif
+
+/**
+ * @brief   List of shell commands for this example.
+ */
+static const shell_command_t shell_commands[] = {
```

Comment:
In general this array is created just before the main function

Reply:
I don't understand this. Can you explain?

Do you mean that I should move this just before the main method?

Reply:
Yes, and then you don't need to declare the command function.

Reply:
Fixed.